Sources/tart/Commands/Run.swift

```diff
@@ -27,24 +27,38 @@ struct Run: AsyncParsableCommand {
           discussion: "Useful since VNC supports copy/paste, drag and drop, etc.\nNote that Remote Login option should be enabled inside the VM.")) 
   var vnc: Bool = false
 
+  @Flag var withSoftnet: Bool = false
+
   @MainActor
   func run() async throws {    
     let vmDir = try VMStorageLocal().open(name)
-    vm = try VM(vmDir: vmDir)
+    vm = try VM(vmDir: vmDir, withSoftnet: withSoftnet)
 
     var vncWrapper: VNCWrapper?
 
     if vnc {
         vncWrapper = VNCWrapper(virtualMachine: vm!.virtualMachine)
     }
 
-    let runTask = Task {
+    let task = Task {
       do {
+        if let vncWrapper = vncWrapper {
+          do {
+            let (port, password) = try await vncWrapper.credentials()
+            let url = URL(string: "vnc://:\(password)@127.0.0.1:\(port)")!
+            print("Opening \(url)...")
+            if ProcessInfo.processInfo.environment["CI"] == nil {
+              NSWorkspace.shared.open(url)
+            }
+          } catch {
+            print("Failed to get an IP for screen sharing: \(error)")
+          }
+        }
```

Comment:
What about putting the opening logic in a `VNCWrapper` method and do something like the following:

```suggestion
        if vnc && ProcessInfo.processInfo.environment["CI"] == nil {
          vncWrapper?.openURL()
        }
```